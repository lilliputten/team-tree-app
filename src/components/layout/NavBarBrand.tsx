'use client';

import Link from 'next/link';
import { usePathname } from 'next/navigation';
import { useLocale } from 'next-intl';

import { TPropsWithChildrenAndClassName } from '@/shared/types/generic';
import { siteConfig } from '@/config/site';
import { getAllRouteSynonyms } from '@/lib/routes';
import { cn } from '@/lib/utils';
import { Logo } from '@/components/shared/Logo';
import { isDev } from '@/constants';
import { infoRoute, welcomeRoute } from '@/constants/routes';
import { TLocale } from '@/i18n/types';

interface NavBarBrandProps {
  isUser: boolean;
  isUserRequired: boolean;
}

function BrandWrapper(props: TPropsWithChildrenAndClassName & NavBarBrandProps) {
  const {
    isUser,
    // isUserRequired,
    children,
    className: parentClassName,
  } = props;
  const locale = useLocale() as TLocale;
  const pathname = decodeURI(usePathname());
  const rootRoute = isUser ? infoRoute : welcomeRoute;
  const rootRoutesList = getAllRouteSynonyms(rootRoute, locale);
  const isRoot = !pathname || rootRoutesList.includes(pathname);
  const className = cn(
    isDev && '__BrandWrapper', // DEBUG
    parentClassName,
    'flex',
    'items-center',
    'space-x-1.5',
    'gap-2',
    'transition-all',
    'mr-10',
    'select-none',
    !isRoot && 'hover:opacity-80',
  );
  if (isRoot) {
    return <div className={className}>{children}</div>;
  }
  return (
    <Link href={rootRoute} className={className}>
      {children}
    </Link>
  );
}

interface NavBarBrandProps {
  isUser: boolean;
  isUserRequired: boolean;
}

export function NavBarBrand(props: NavBarBrandProps) {
  return (
    <BrandWrapper {...props} className="h-12">
      <Logo className="size-8" />
      <h1
        role="heading"
        data-testid="NavBarBrandTitle"
        className={cn(
          'font-urban',
          'text-xl',
          'text-primary-foreground',
          'font-bold',
          'whitespace-nowrap',
          'overflow-hidden',
          'text-ellipsis',
        )}
      >
        {siteConfig.name}
      </h1>
    </BrandWrapper>
  );
}
