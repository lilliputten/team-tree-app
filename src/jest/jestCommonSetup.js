/* eslint-env jest */

/** Test db. Ensure if it has been created */
const DATABASE_URL = 'file:.data/test.db';

/* NOTE: The `@t3-oss/env-nextjs` module ruins the tests with weird
 * `SyntaxError: Cannot use import statement outside a module`. It should be
 * mocked (see `jestCommonSetup.js`).
 */

jest.mock('@t3-oss/env-nextjs', () => ({
  createEnv: jest.fn(() => ({
    DATABASE_URL: DATABASE_URL,
    NEXT_PUBLIC_APP_URL: 'http://localhost:3000',
  })),
}));

// Set it for prisma also
process.env.DATABASE_URL = DATABASE_URL;
