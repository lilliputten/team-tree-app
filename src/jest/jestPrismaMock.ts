import { DeepMockProxy, mockDeep, mockReset } from 'jest-mock-extended';
import { PrismaClient } from '@prisma/client';

// import prisma from './client'
// import { prisma } from '@/lib/db/prisma';
import { jestPrisma } from '@/lib/db/jestPrisma';

jest.mock('@/lib/db/prisma', () => ({
  __esModule: true,
  default: mockDeep<PrismaClient>(),
}));

beforeEach(() => {
  console.log('[jestPrismaMock] 2', {
    // prisma,
    jestPrisma,
    jestPrismaMock,
    mockReset,
    mockDeep,
    // DeepMockProxy,
  });
  debugger;
  mockReset(jestPrismaMock);
});

export const jestPrismaMock = jestPrisma as unknown as DeepMockProxy<PrismaClient>;
console.log('[jestPrismaMock] 1', {
  // prisma,
  jestPrisma,
  jestPrismaMock,
  mockReset,
});
