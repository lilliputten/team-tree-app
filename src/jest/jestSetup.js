// Learn more: https://github.com/testing-library/jest-dom
// https://nextjs.org/docs/app/building-your-application/testing/jest

import '@testing-library/jest-dom';

// process.env.DATABASE_URL = 'file:.data/test.db';
