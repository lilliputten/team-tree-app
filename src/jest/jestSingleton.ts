import { DeepMockProxy, mockDeep, mockReset } from 'jest-mock-extended';
import { PrismaClient } from '@prisma/client';

import { jestPrisma } from '@/lib/db/jestPrisma';

jest.mock('@/lib/db/prisma', () => ({
  __esModule: true,
  default: mockDeep<PrismaClient>(),
}));

beforeEach(() => {
  mockReset(prismaMock);
});

export const prismaMock = jestPrisma as unknown as DeepMockProxy<PrismaClient>;
