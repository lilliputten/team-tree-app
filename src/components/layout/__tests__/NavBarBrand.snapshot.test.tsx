/** @jest-environment jsdom
 */

import '@/jest/jestDomSetup';

import { render } from '@testing-library/react';

import { NavBarBrand } from '../NavBarBrand';

it('NavBarBrand snapshot', () => {
  const { container } = render(<NavBarBrand />);
  expect(container).toMatchSnapshot();
});
