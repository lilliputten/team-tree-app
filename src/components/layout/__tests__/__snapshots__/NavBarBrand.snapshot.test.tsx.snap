// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarBrand snapshot 1`] = `
<div>
  <a
    class="h-12 flex items-center space-x-1.5 gap-2 transition-all mr-10 select-none hover:opacity-80"
    href="/welcome"
  >
    <img
      alt="Logo"
      class="size-8"
      data-nimg="1"
      decoding="async"
      height="36"
      loading="lazy"
      src="/static/logo/LogoWhite-min.svg"
      style="color: transparent;"
      width="36"
    />
    <h1
      class="font-urban text-xl text-primary-foreground font-bold whitespace-nowrap overflow-hidden text-ellipsis"
      data-testid="NavBarBrandTitle"
      role="heading"
    >
      Team Tree
    </h1>
  </a>
</div>
`;
